src/test/unit/services/sample/standalone_gqs_test.cpp

```diff
@@ -2,159 +2,126 @@
+                           std::fstream::in);
+  stan::io::dump data_var_context(data_stream);
+  data_stream.close();
+  model = new stan_model(data_var_context);
+ }
+
+  void TearDown() {
+    delete(model);
+  }
 
-  stan::io::empty_var_context context;
-  std::stringstream model_log;
-  stan_model model;
-  std::stringstream sample_ss, logger_ss;
   stan::test::unit::instrumented_interrupt interrupt;
+  std::stringstream logger_ss;
+  stan::callbacks::stream_logger logger;
+  stan_model *model;
 };
 
-TEST_F(ServicesStandaloneGQ, numParams) {
-  // model test_gq.stan has 2 params
-  int num_params = stan::services::num_constrained_params(model);
-  EXPECT_EQ(num_params, 2);
-}
+TEST_F(ServicesStandaloneGQ, genDraws_bernoulli) {
+  stan::io::stan_csv bern_csv;
+  std::stringstream out;
+  std::ifstream csv_stream;
+  csv_stream.open("src/test/test-models/good/services/bernoulli_fit.csv");
+  bern_csv = stan::io::stan_csv_reader::parse(csv_stream, &out);
+  csv_stream.close();
+  EXPECT_EQ(12345U, bern_csv.metadata.seed);
+  ASSERT_EQ(19, bern_csv.header.size());
+  EXPECT_EQ("theta", bern_csv.header(7));
+  ASSERT_EQ(1000, bern_csv.samples.rows());
+  ASSERT_EQ(19, bern_csv.samples.cols());
+
+  // model bernoulli.stan has 1 param
+  std::vector<std::string> param_names;
+  std::vector<std::vector<size_t>> param_dimss;
+  stan::services::get_model_parameters(*model, param_names, param_dimss);
 
-TEST_F(ServicesStandaloneGQ, genDraws_good) {
-  std::vector<double> draw1;
-  draw1.push_back(-2.345);
-  draw1.push_back(-6.789);
-  std::vector<double> draw2;
-  draw2.push_back(-3.123);
-  draw2.push_back(-4.123);
-  std::vector<std::vector<double> > draws;
-  draws.push_back(draw1);
-  draws.push_back(draw2);
-  const std::vector<std::vector<double> > cdraws(draws);
+  EXPECT_EQ(param_names.size(), 1);
+  EXPECT_EQ(param_dimss.size(), 1);
+
+  std::stringstream sample_ss;
   stan::callbacks::stream_writer sample_writer(sample_ss, "");
-  stan::callbacks::stream_logger logger(logger_ss,
-                                        logger_ss,
-                                        logger_ss,
-                                        logger_ss,
-                                        logger_ss);
-  int return_code = stan::services::standalone_generate(model,
-                                                        cdraws,
+  int return_code = stan::services::standalone_generate(*model,
+                                                        bern_csv.samples.middleCols<1>(7),
                                                         12345,
                                                         interrupt,
                                                         logger,
                                                         sample_writer);
-  // model test_gq.stan gen quantities block has 3 params:  xqg, y_rep.1, y_rep.2
+  if (return_code != stan::services::error_codes::OK)
+    std::cout << "ERROR: " << logger_ss.str() << std::endl;
```

Comment:
(all of these need to assert failure)

Reply:
all of them test the return code:

`EXPECT_EQ(return_code, stan::services::error_codes::<appropriate-code-for-test>);`